src/lib.rs

```diff
@@ -288,8 +290,11 @@ where
             .map_err(|error| Error::command_io_error(&config, error))?;
     }
     let mut command = Command::new(&executable);
+    command.args(arguments);
+    for (key, value) in &config.environment_additions {
+        command.env(key, value);
+    }
     command
-        .args(arguments)
         .stdin(Stdio::piped())
```

Comment:
I haven't tested this, but I think it should work:

```suggestion
        .args(arguments)
        .envs(&config.environment_additions)
        .stdin(Stdio::piped())
```

Reply:
I couldn't make this work. `Command::envs` expects an iterator over tuples, not references to tuples. So we could clone the tuple, of course. But that seems worse, since with the for loop above, we don't have to clone anything.